ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Monitor.hs

```diff
@@ -144,14 +144,19 @@ connections PeerSelectionActions{
               inProgressDemoteHot,
               inProgressDemoteWarm,
               inProgressPromoteWarm,
+              inProgressDemoteToCold,
               fuzzRng
             } =
     Guarded Nothing $ do
+      -- Get previously cooling peers
       monitorStatus <- traverse monitorPeerConnection
                                 (EstablishedPeers.toMap establishedPeers)
+          -- filter previously cooling peers from the demotion sets
       let demotions = asynchronousDemotions monitorStatus
+                      `Map.withoutKeys` inProgressDemoteToCold
       check (not (Map.null demotions))
-      let (demotedToWarm, demotedToCold) = Map.partition ((==PeerWarm) . fst) demotions
+      let (demotedToWarm, demotedToCoolingOrCold) = Map.partition ((==PeerWarm) . fst) demotions
```

Comment:
It would be but I don't actually need a `demotedTooCooling` set

Reply:
And I suspect we also need the `demotedToCold` because we want subtract them from `inProgressDemoteToCold`.

Reply:
It would be good to also leave a comment somewhere that it's not possible for a peer to be promoted when it's in `PeerCooling` state (that's the reason for the new state), so `PeerCold` is the only state through which peers are removed from `inProgressDemoteToCold`.

Reply:
Will do!